src/components/Modal/VotingPowerDelegationDetail/VotingPowerDelegationHandler.tsx

```diff
@@ -14,36 +14,43 @@ interface Props {
   userVP: number
 }
 
-function VotingPowerDelegationButton({ candidateAddress, userVP }: Props) {
+function VotingPowerDelegationHandler({ candidateAddress, userVP }: Props) {
   const t = useFormatMessage()
   const [isLoading, setLoading] = useState(false)
+  const [isError, setIsError] = useState(false)
   const { isContractUsable, delegatedAddress, isGlobalDelegation, setDelegate, clearDelegate } =
     useSnapshotDelegateContract()
 
   const isRevocable = !!delegatedAddress && delegatedAddress.toLowerCase() === candidateAddress.toLowerCase()
   const isDelegatedGlobally = isRevocable && isGlobalDelegation
+  const showDelegatedInfo = isRevocable && !isLoading && !isError
 
   const handleClick = async () => {
     setLoading(true)
-
+    setIsError(false)
     try {
       if (isRevocable) {
         await clearDelegate()
       } else {
         await setDelegate(candidateAddress)
       }
-    } catch (error) {
+    } catch (error: any) {
       console.error(error)
+      if (error.code !== 4001) { // transaction canceled by user
```

Comment:
I'd move 4001 to a constant TRANSACTION_CANCELED_BY_USER_STATUS_CODE or something like that and remove the comment.